libs/langchain-community/src/vectorstores/azure_cosmosdb.ts

```diff
@@ -179,10 +200,17 @@ export class AzureCosmosDBVectorStore extends VectorStore {
    */
   async createIndex(
     numLists = 100,
-    dimensions = 1536,
+    dimensions: number | undefined = undefined,
     similarity: AzureCosmosDBSimilarityType = AzureCosmosDBSimilarityType.COS
   ): Promise<void> {
-    await this.initPromise;
+    await this.connectPromise;
```

Comment:
If this is hypothetically called before `connectPromise` is initialized, could we throw instead?

Reply:
Ah never mind, I see below. Could we always just rely on `this.initPromise` instead?

Reply:
Since `init` calls `createIndex` we need 2 different promise there otherwise there's an interlocking:
- connectPromise => create DB + collections clients
- initPromise => connectPromise + createIndex

Reply:
For the throw question, the issue is that the promise is created in the constructor and there's no way for the user to wait for it or know when the connect task is done.